From c93580ca8eb05630483dddbfcbb886a9af09476b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A9s=20Fern=C3=A1ndez?=
 <andresjusticialista@gmail.com>
Date: Tue, 10 Mar 2026 21:07:24 -0300
Subject: [PATCH 1/2] application.vala override startup

Source: https://gitlab.com/softwareperonista/rockarrolla/-/merge_requests/124/diffs

This method will run base startup method before accesing to Adw.StyleManager
---
 src/application.vala | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/src/application.vala b/src/application.vala
index cbf3fba..2b795f2 100644
--- a/src/application.vala
+++ b/src/application.vala
@@ -19,14 +19,20 @@
 namespace Rockarrolla {
   public class Application : Adw.Application {
     public Application() {
-      Object(application_id: "ar.com.softwareperonista.Rockarrolla", flags : ApplicationFlags.FLAGS_NONE);
+      Object(
+             application_id: "ar.com.softwareperonista.Rockarrolla",
+             flags: ApplicationFlags.FLAGS_NONE
+      );
+    }
+
+    protected override void startup() {
+      base.startup();
 
-      var style_manager = this.get_style_manager();
+      var style_manager = Adw.StyleManager.get_default();
       style_manager.set_color_scheme(Adw.ColorScheme.PREFER_DARK);
     }
 
     public override void activate() {
-      base.activate();
       var win = this.active_window;
       if (win == null) {
         win = new Rockarrolla.Window(this);
-- 
GitLab


From c03f74e2e53ed7132c7007dfd8c3064f0f583049 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A9s=20Fern=C3=A1ndez?=
 <andresjusticialista@gmail.com>
Date: Tue, 10 Mar 2026 21:16:07 -0300
Subject: [PATCH 2/2] meson.build: updated minimum meson version required

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 44276c1..01e0f7e 100644
--- a/meson.build
+++ b/meson.build
@@ -2,7 +2,7 @@ project(
   'rockarrolla',
   ['c', 'vala'],
   version: '1.4',
-  meson_version: '>= 0.49.0',
+  meson_version: '>= 0.54.0',
 )
 
 i18n = import('i18n')
-- 
GitLab

